docs/en/cpp/guide/mavlink_direct.md

```diff
@@ -0,0 +1,226 @@
+# MavlinkDirect Plugin Guide
+
+The [MavlinkDirect](../cpp/api_reference/classmavsdk_1_1_mavlink_direct.md) plugin enables direct MAVLink communication with runtime message parsing and JSON field representation.
+
+::: warning
+The MavlinkDirect API is quite new and is not necessarily stabilized yet.
+The functionality is here to stay but the specifics and types might still change before MAVSDK v4.
+:::
+
+::: tip
+MavlinkDirect is the recommended replacement for [MavlinkPassthrough](../cpp/api_reference/classmavsdk_1_1_mavlink_passthrough.md).
+In MAVSDK v4, MavlinkPassthrough will only be available as a compile-time option.
+:::
+
+::: info
+The MavlinkDirect plugin is built on top of the work by Thomas Debrunner in [libmav](https://github.com/Auterion/libmav).
+:::
+
+## Overview
+
+MavlinkDirect allows you to:
+
+- Send and receive any MAVLink message
+- Load custom MAVLink message definitions from XML
+
+## Runtime vs compile-time considerations
+
+MavlinkDirect works at runtime instead of compile-time, so instead of having the C types and functions to rely on, everything has to be done using string methods. There are trade-offs with that:
+
+**Runtime pros:**
+
+- Allows extension with custom messages/dialects without re-compiling MAVSDK.
+- Allows usage in language wrappers such as Python.
+- MAVSDK no longer needs to carry and install the MAVLink C headers.
+
+**Runtime cons:**
+
+- API users need to look up MAVLink message definitions and don't get auto-complete.
+- Enum values or flags/bits need to be assembled manually.
+- The API is likely slower although that's likely negligible in C++ as long as messages are not sent/received at very high rates (benchmarks outstanding).
+
+## Getting Started
```

Comment:
There is an inconsistency in heading capitalization across this doc. Doesn't matter which you use, but be consistent. I think the docs broadly use the format "Basic obstacle example" but I haven't checked